shell/edit/auth/oidc.vue

```diff
@@ -58,43 +67,64 @@ export default {
         oidcConfig: this.model
       };
     },
-
   },
+
   watch: {
-    'keycloakUrls.url'() {
-      this.updateIssuerEndpoint();
+    'oidcUrls.url'() {
+      this.updateEndpoints();
     },
-    'keycloakUrls.realm'() {
-      this.updateIssuerEndpoint();
+
+    'oidcUrls.realm'() {
+      this.updateEndpoints();
     },
+
     'model.enabled'(neu) {
       // Cover case where oidc gets disabled and we return to the edit screen with a reset model
       if (!neu) {
-        this.keycloakUrls = {
-          url:   null,
-          realm: null
+        this.oidcUrls = {
+          url:              null,
+          realm:            null,
+          jwksUrl:          null,
+          tokenEndpoint:    null,
+          userInfoEndpoint: null,
         };
         this.customEndpoint.value = false;
+        this.oidcScope = [];
+      } else {
+        this.oidcScope = this.model?.scope?.split(' ');
       }
     },
+
     editConfig(neu, old) {
-      // Cover use case where user edits existing oidc (keycloakUrls aren't persisted, so if we have issuer & authEndpoint set custom endpoints to true)
+      // Cover use case where user edits existing oidc (oidcUrls aren't persisted, so if we have issuer & authEndpoint set custom endpoints to true)
       if (!old && neu) {
-        this.customEndpoint.value = (!this.keycloakUrls.url && !this.keycloakUrls.authEndpoint) && (!!this.model.issuer && !!this.model.authEndpoint);
+        this.customEndpoint.value = (!this.oidcUrls.url && !this.oidcUrls.authEndpoint) && (!!this.model.issuer && !!this.model.authEndpoint);
       }
     }
   },
 
   methods: {
-    updateIssuerEndpoint() {
-      if (!this.keycloakUrls.url) {
+    updateEndpoints() {
+      if (!this.oidcUrls.url) {
         return;
       }
-      const url = this.keycloakUrls.url.replaceAll(' ', '');
+      const url = this.oidcUrls.url.replaceAll(' ', '');
 
-      this.model.issuer = `${ url }/auth/realms/${ this.keycloakUrls.realm || '' }`;
+      this.model.issuer = `${ url }/auth/realms/${ this.oidcUrls.realm || '' }`;
```

Comment:
In Chad's test instance the OIDC urls don't contain `auth`
- Issuer - `.../auth/realms/testrealm` --> `.../realms/testrealm`
- Auth Endpoint - `..../auth/realms/testrealm/protocol/openid-connect/auth` --> `..../realms/testrealm/protocol/openid-connect/auth`

@crobby what would be sensible defaults to use for generic oidc urls, is your instance typical and the `auth` in the path is more of a keycloak thing?

Reply:
I think the paths in my sample keycloak instance are a bit of "a keycloak thing"
I'm not sure what the most sensible default would be.  Let me poke around a bit.

Reply:
Maybe it makes the most sense to have no default.  The generic oidc implementation will perform "discovery" based on the Issuer if the other endpoints are not provided, so defaulting it to empty may make the most sense.  From what I've seen, having discovery turned on is the most typical configuration.

Reply:
@crobby ATM the UI requests the root url and realm, and then constructs mandatory `issuer` and `authEndpoint` urls.  

- We would need at least the issuer url
- Is the authEnpoint mandatory, or does discovery work from just the issuer?

I get the feeling we should just go with the removal of `auth` from the urls. The generate step is designed to get the user quickly to the point of discovery. If they don't work then the user can choose to `Specify` and tweak our suggestions, and then optional avoid discovery by adding more advanced urls

Reply:
Auth is not mandatory for the generic provider.  If it's not present, we'll use the issuer to perform discovery for all the other endpoints.
I like the idea of removing auth from the "generate" step.